src/adapters/uwulend-staking.adapter.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright Abridged, Inc. 2022. All Rights Reserved.
+// Node module: @collabland/staking-contracts
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking';
+import {UwulendStaking__factory} from '../types/factories/uwulendstaking__factory';
+
+@injectable(
+  {
+    scope: BindingScope.SINGLETON,
+  },
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class UwULendStakingContractAdapter extends BaseStakingContractAdapter {
+  contractAddress = '0xe3643512532Fe1f3522745787e883F9729527186';
+
+  supportedAssets: StakingAsset[] = [
+    {
+      asset: 'ERC20:0x3E04863DBa602713Bb5d0edbf7DB7C3A9A2B6027',
+    },
+  ];
+
+  getStakedTokenBalance(owner: string): Promise<BigNumber> {
+    const contract = UwulendStaking__factory.connect(
+      this.contractAddress,
+      this.provider,
+    );
+    return contract.balanceOf(owner);
```

Comment:
Since the contract implements `balanceOf` view method, do we still need to add it as a staking contract? It should be possible to use it as a regular ERC20 contract.

Reply:
I'm not aware of why but it doesn't seems to work.

Reply:
I meant to say your can define a TGR using the staking contract as regular ERC20.

Reply:
yeah we've done that, it doesn't detect the users.